src/art.rs

```diff
@@ -1171,66 +1148,17 @@ impl<P: KeyTrait, V: Clone> Tree<P, V> {
     /// Returns a `Result` containing the `Snapshot` if the snapshot is created successfully,
     /// or an `Err` with an appropriate error message if creation fails.
     ///
-    pub fn create_snapshot(&mut self) -> Result<Snapshot<P, V>, TrieError> {
+    pub fn create_snapshot(&self) -> Result<Snapshot<P, V>, TrieError> {
         // Check if the tree is already closed
-        self.is_closed()?;
```

Comment:
Shouldnt this check stay?

Reply:
You are right, I have actually just changed the name